pkg/util/iptables/iptables.go

```diff
@@ -235,13 +229,22 @@ func newInternal(exec utilexec.Interface, protocol Protocol, lockfilePath14x, lo
 	runner := &runner{
 		exec:            exec,
 		protocol:        protocol,
-		hasCheck:        version.AtLeast(MinCheckVersion),
-		hasRandomFully:  version.AtLeast(RandomFullyMinVersion),
-		waitFlag:        getIPTablesWaitFlag(version),
-		restoreWaitFlag: getIPTablesRestoreWaitFlag(version, exec, protocol),
 		lockfilePath14x: lockfilePath14x,
 		lockfilePath16x: lockfilePath16x,
 	}
+
+	version, err := getIPTablesVersion(exec, protocol)
+	if err != nil {
```

Comment:
I still think that logging the original error is valuable to debug weird problems, is not likely that this happen but when this will happen not having information like this will be very complex to understand .. I'm thinking in things like per example the iptables binary is not executable or has a different architecture that the process this is running, I know is very niche but that is also a point why logging here will not be expensive

Reply:
I should have read better the commit description is explained there, sorry, it was already weird before,

Reply:
We could change New/NewDualStack to return an error... (`knftables.New()` does, and we log that error if it occurs.) I just wasn't sure if we cared...

Reply:
It looks more correct to me, but to be honest this was this way forever and I can't remember an issue related